transcom/mymove
# [MB-16548] Import TRDM data

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16548)

## Summary

Created a secure migration to import TAC and LOA data.
I imported both into separate tables and then set up the foreign keys so that TAC records are linked to LOA records by the `loa_sys_id` field.
I deleted all the old entries in the TAC table.

There was so much TAC data, I had to split it into 3 migrations because I was unable to upload files >250MB.

I also had to modify my favorite part of the code, the custom SQL parser we use for our migrations.
Double dashes (`--`) in the body of `COPY` statements were getting interpreted as comments, so the rest of the line would be dropped.
Similar deal with the `DO`. When its in the data for a COPY, it should just be intrepreted as text, but it got parsed as a SQL keyword and throws things off.

Also, `-- POP RAW MIGRATION --` didn't work for my scenario, and I couldn't figure out why. Seems like maybe it just doesn't work with copy from stdin.

See this confluence page for more information about the schema: https://dp3.atlassian.net/wiki/spaces/MT/pages/2194571273/Discovery+for+MB-15743

Let me know if I can tweak the test data to be more useful for followup work!

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.
### How to test
First, make sure the local migration with dummy data works
```
make db_dev_reset
make db_dev_migrate
```

Then make sure that the secure migration with the real data works
- Instructions for downloading secure migrations can be found here: https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#downloading-secure-migrations

It would also be good to download the original data files from Keeper and make sure that they match up with the database tables that get created


### Database

#### Any new migrations/schema changes:
